nipoppy_cli/nipoppy/config/pipeline.py

```diff
@@ -2,70 +2,114 @@
 
 from __future__ import annotations
 
-import re
 from pathlib import Path
-from typing import Optional, Sequence
+from typing import Optional
 
-from pydantic import ConfigDict, model_validator
+from pydantic import ConfigDict, Field, model_validator
 
-from nipoppy.config.container import ModelWithContainerConfig
+from nipoppy.config.container import ContainerInfo, SchemaWithContainerConfig
+from nipoppy.config.pipeline_step import PipelineStepConfig
 
 
-class PipelineConfig(ModelWithContainerConfig):
-    """Model for workflow configuration."""
+class PipelineConfig(SchemaWithContainerConfig):
+    """Schema for processing pipeline configuration."""
 
-    DESCRIPTION: Optional[str] = None
-    CONTAINER: Optional[Path] = None
-    URI: Optional[str] = None
-    DESCRIPTOR: Optional[dict] = None
-    DESCRIPTOR_FILE: Optional[Path] = None
-    INVOCATION: Optional[dict] = None
-    INVOCATION_FILE: Optional[Path] = None
-    PYBIDS_IGNORE: list[re.Pattern] = []
-    TRACKER_CONFIG: dict[str, list[str]] = {}
+    NAME: str = Field(description="Name of the pipeline")
+    VERSION: str = Field(description="Version of the pipeline")
+    DESCRIPTION: Optional[str] = Field(
+        default=None, description="Free description field"
+    )
+    CONTAINER_INFO: ContainerInfo = Field(
+        default=ContainerInfo(),
+        description="Information about the container image file",
+    )
+    STEPS: list[PipelineStepConfig] = Field(
+        default=[],
+        description="List of pipeline step configurations",
+    )
+    TRACKER_CONFIG_FILE: Optional[Path] = Field(
+        default=None,
+        description=(
+            "Path to the tracker configuration file associated with the pipeline"
+            ". This file must contain a list of tracker configurations"
+            ", each of which must be a dictionary with a NAME field (string)"
+            " and a PATHS field (non-empty list of strings)"
+        ),
+    )
 
     model_config = ConfigDict(extra="forbid")
 
     @model_validator(mode="after")
-    def check_fields(self):
+    def validate_after(self):
         """
-        Check that <FIELD> and <FIELD>_FILE fields are not both set.
+        Validate the pipeline configuration after creation.
 
-        Also add an empty invocation if none is provided.
+        Specifically:
+        - If STEPS has more than one item, make sure that each step has a unique name.
```

Comment:
Will it be easier to have a generic "duplication" check method in workflow that does this for multiple things e.g. pipelines, versions, steps, trackers etc.. ?